usr/lib/pkcs11/common/sw_crypt.c

```diff
@@ -74,3 +75,35 @@ CK_RV sw_des3_cbc(CK_BYTE *in_data,
 
     return CKR_OK;
 }
+
+CK_RV sw_aes_cbc(CK_BYTE *in_data,
+                 CK_ULONG in_data_len,
+                 CK_BYTE *out_data,
+                 CK_ULONG *out_data_len,
+                 CK_BYTE *init_v, CK_BYTE *key_value, CK_ULONG keylen,
+                 CK_BYTE encrypt)
+{
+    AES_KEY aes_key;
+
+    memset(&aes_key, 0, sizeof(aes_key));
+
+    // the aes decrypt will only fail if the data length is not evenly divisible
+    // by AES_BLOCK_SIZE
+    if (in_data_len % AES_BLOCK_SIZE) {
+        TRACE_ERROR("%s\n", ock_err(ERR_DATA_LEN_RANGE));
+        return CKR_DATA_LEN_RANGE;
+    }
+
+    // Encrypt or decrypt the data
+    if (encrypt) {
+        AES_set_encrypt_key(key_value, keylen * 8, &aes_key);
```

Comment:
Not using evp interfaces means no guarantee for full platform specific hardware support (old cbc mode apis have full hw support on z though). Dont know if encrypting/decrypting master key is relevant wrt performance.

Reply:
I don't think is performance relevant. this code is only used for master key encryption, not for real data encryption. Real data encryption uses the token specific crypto functions.

Reply:
okay im fine with it.